jina/drivers/convert.py

```diff
@@ -136,7 +139,35 @@ def png_pack(png_tag, data):
 
         return 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
 
-    def convert(self, arr):
+    @staticmethod
+    def image_to_byte_array(image: 'Image', format: str):
+        import io
+        img_byte_arr = io.BytesIO()
+        image.save(img_byte_arr, format=format)
+        img_byte_arr = img_byte_arr.getvalue()
+        return img_byte_arr
+
+    def png_convertor(self, arr: np.array):
+        from PIL import Image
+
+        arr = arr.astype(np.uint8)
+
+        if len(arr.shape) == 1:
+            # im = Image.fromarray(arr.reshape((self.height, self.width))).convert('L')
```

Comment:
remove comment

Reply:
This is our replacement code for the png_converter_1d. Will remove and save it somewhere